### PR TITLE
Cypher type name parser

### PR DESCRIPTION
**Pull request by DarthMax**
Adds `CypherType.fromName("INTEGER") -> Some(CTInteger)`

### CODE REVIEW

File: okapi-api/src/main/scala/org/opencypher/okapi/api/types/CypherType.scala [hunk 1]
```diff
@@ -34,6 +34,58 @@ import scala.language.postfixOps
...
+      case "PATH"    | "PATH?"    => Some(CTPath)
+      case "?"       | "??"       => Some(CTWildcard)
+
+      case node if node.startsWith("NODE") =>
+        Some(CTNode(extractLabels(node, "NODE", ":"), extractGraph(node)))
+
+      case rel if rel.startsWith("RELATIONSHIP") =>
+        Some(CTRelationship(extractLabels(rel, "RELATIONSHIP", """\|"""), extractGraph(rel)))
+
+      case list if list.startsWith("LIST") =>
```
Comment (pstutz) [inline okapi-api/src/main/scala/org/opencypher/okapi/api/types/CypherType.scala]: Does this approach work for nested structures?

File: okapi-api/src/test/scala/org/opencypher/okapi/api/types/CypherTypesTest.scala [hunk 2]
```diff
@@ -115,6 +115,12 @@ class CypherTypesTest extends FunSpec with Matchers {
     CTNull.toString shouldBe "NULL"
   }
 
+  it("can parse CypherType names into CypherTypes"){
+    allTypes.foreach { t =>
+      CypherType.fromName(t.name).get should equal(t)
```
Comment (pstutz) [inline okapi-api/src/test/scala/org/opencypher/okapi/api/types/CypherTypesTest.scala]: Did you test the graph extraction?

**Comment by pstutz** [comment]
Thanks a lot! 👍 I think this might be useful elsewhere and it's not easy to get right.

File: okapi-api/src/main/scala/org/opencypher/okapi/api/types/CypherType.scala (hunk 1 above)
Comment (DarthMax) [reply okapi-api/src/main/scala/org/opencypher/okapi/api/types/CypherType.scala]: There are tests with Lists in the test suite. I think it should work

File: okapi-api/src/test/scala/org/opencypher/okapi/api/types/CypherTypesTest.scala (hunk 2 above)
Comment (DarthMax) [reply okapi-api/src/test/scala/org/opencypher/okapi/api/types/CypherTypesTest.scala]: good point

**Pull request merged**